adoc/chapters/programming_interface.adoc

```diff
@@ -10165,45 +10191,48 @@ public:
 a@
 [source]
 ----
-usm_allocator(const context &ctxt,
-              const device &dev,
+usm_allocator(const context &syclContext,
+              const device &syclDevice,
               const property_list &propList = {})
 ----
-a@ Constructs a [code]#usm_allocator# instance that allocates USM memory for
-the provided context and device.  If [code]#AllocKind# is
-[code]#usm::alloc::host#, the [code]#dev# parameter is ignored.  For other
-values of [code]#AllocKind#, [code]#dev# must either be contained by
-[code]#ctxt# or it must be a <<descendent-device>> of some device that is
-contained by [code]#ctxt#, otherwise this constructor throws a synchronous
-[code]#exception# with the [code]#errc::invalid# error code.
+a@ Constructs a [code]#usm_allocator# instance that allocates USM for the
+provided context and device.
+
+If [code]#AllocKind# is [code]#usm::alloc::host#, this constructor throws a
+synchronous [code]#exception# with the [code]#errc::feature_not_supported#
+error code if no device in [code]#syclContext# has
+[code]#aspect::usm_host_allocations#.  The [code]#syclDevice# is ignored for
+this allocation kind.
```

Comment:
This is a good point.  If I were doing this from scratch, I would propose different constructors depending on the template parameter `AllocKind`:

```
// When "AllocKind == usm::alloc::shared"
usm_allocator(const context& syclContext, const device& syclDevice, const property_list& propList = {});

// When "AllocKind == usm::alloc::host"
usm_allocator(const context& syclContext, const property_list& propList = {});
```

If we make this change now, though, we risk breaking existing code that is written to the current spec.

Depending on how others feel, we could still consider making this change with the following points in mind:

* There is probably not a lot of code that uses `usm_allocator` today because this seems like a more obscure feature.
* Implementations like DPC++ could continue to support the existing constructor form for "AllocKind == usm::alloc::host", but mark it as deprecated.

Reply:
Your last point looks safe.

Reply:
I created internal issue 622 to track this issue about the constructor form for `AllocKind == usm::alloc::host`.  I think we have decided not to make that change as part of this PR.